<launch>
    <arg name="enable_rviz" default="true" />
    <!-- 0：读取配置文件中的轨迹 1: 实时计算 -->
    <arg name="dynamic_type" default="1" />
    <!-- agent name -->
    <arg name="agent_name" default="ugv" />
    <!-- 集群数量 -->
    <arg name="agent_num" default="3" />
    <!-- agent类型 0：uav 1: ugv -->
    <arg name="agent_type" default="1" />
    <arg name="file_path" default="/home/yundrone/Sunray/sunray_formation/uav_formation/config/config.yaml" />
    <arg name="circle/lissa_a" default="1" />
    <arg name="circle/lissa_b" default="1" />
    <arg name="circle/lissa_delta" default="1.5707" />
    <arg name="circle/radius_x" default="1.5" />
    <arg name="circle/radius_y" default="1.5" />
    <arg name="circle/center_x" default="0" />
    <arg name="circle/center_y" default="0" />
    <arg name="circle/linear_speed" default="0.25" />
    <arg name="figure_eight/lissa_a" default="1" />
    <arg name="figure_eight/lissa_b" default="2" />
    <arg name="figure_eight/lissa_delta" default="1.5707" />
    <arg name="figure_eight/radius_x" default="2.5" />
    <arg name="figure_eight/radius_y" default="0.8" />
    <arg name="figure_eight/center_x" default="0" />
    <arg name="figure_eight/center_y" default="0" />
    <arg name="figure_eight/linear_speed" default="0.25" />
    
    <node name="uav_formation_node_1" pkg="sunray_formation" type="formation_control_node" output="screen">
        <param name="agent_id" type="int" value="1" />
        <param name="agent_name" type="string" value="$(arg agent_name)" />
        <param name="agent_num" type="int" value="$(arg agent_num)" />
        <param name="agent_type" type="int" value="$(arg agent_type)" />
        <param name="file_path" type="string" value="/home/yundrone/Sunray/sunray_formation/uav_formation/config/config.yaml" />
        <param name="dynamic_type" type="int" value="$(arg dynamic_type)" />
        <param name="circle/lissa_a" type="double" value="$(arg circle/lissa_a)" />
        <param name="circle/lissa_b" type="double" value="$(arg circle/lissa_b)" />
        <param name="circle/lissa_delta" type="double" value="$(arg circle/lissa_delta)" />
        <param name="circle/radius_x" type="double" value="$(arg circle/radius_x)" />
        <param name="circle/radius_y" type="double" value="$(arg circle/radius_y)" />
        <param name="circle/center_x" type="double" value="$(arg circle/center_x)" />
        <param name="circle/center_y" type="double" value="$(arg circle/center_y)" />
        <param name="circle/linear_speed" type="double" value="$(arg circle/linear_speed)" />
        <param name="figure_eight/lissa_a" type="double" value="$(arg figure_eight/lissa_a)" />
        <param name="figure_eight/lissa_b" type="double" value="$(arg figure_eight/lissa_b)" />
        <param name="figure_eight/lissa_delta" type="double" value="$(arg figure_eight/lissa_delta)" />
        <param name="figure_eight/radius_x" type="double" value="$(arg figure_eight/radius_x)" />
        <param name="figure_eight/radius_y" type="double" value="$(arg figure_eight/radius_y)" />
        <param name="figure_eight/center_x" type="double" value="$(arg figure_eight/center_x)" />
        <param name="figure_eight/center_y" type="double" value="$(arg figure_eight/center_y)" />
        <param name="figure_eight/linear_speed" type="double" value="$(arg figure_eight/linear_speed)" />
    </node>

    <group if="$(arg enable_rviz)">
        <arg name="rivz_config" default="$(find sunray_uav_control)/rviz/formation.rviz"/>
        <node type="rviz" name="rviz_external_fusion" pkg="rviz" args="-d $(arg rivz_config)"/>
    </group>

</launch>
